From 6ceffedac405ccc5fc89e198cff5c9124c2afc55 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Sun, 5 May 2024 12:06:00 +0300
Subject: [PATCH 1/2] compositor: render full screen after startup
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

There is a history of artifacts after startup. Commit 554e5c7a
was applied to clear the frame buffer so that "undefined" data
would not be used. That "undefined" data often was latest rendered
image, and it was actually masking the underlying problem.

On startup, a full paint of the screen is never scheduled for
the frame buffer. This causes parts of the render buffer to remain
untouched (now black). And when executing a flip, those areas are
now black instead of "undefined" or "likely old image".

The problem is clearly visible, when you have background set, and you
execute xfwm4 --replace from a terminal. Most of the background will
be black instead of the background image. When moving the terminal
around it will "fix" the areas it moves over.

Another visible glitch is flash of black screen when logging in
from lightdm or other display manager which preloads the users
background image as root tile before starting the window manager.

Fix this be scheduling a full rendering of the screen once it
becomes managed by the compositor.

Signed-off-by: Timo Teräs <timo.teras@iki.fi>
Closes: https://gitlab.xfce.org/xfce/xfwm4/-/issues/684
---
 src/compositor.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/compositor.c b/src/compositor.c
index 9b92db921..74fed8731 100644
--- a/src/compositor.c
+++ b/src/compositor.c
@@ -4977,6 +4977,8 @@ compositorManageScreen (ScreenInfo *screen_info)
                              screen_info->xroot,
                              XFixesDisplayCursorNotifyMask);
 
+    damage_screen (screen_info);
+
     return TRUE;
 #else
     return FALSE;
-- 
GitLab


From 2c44cd64996b3cb45e16147713c31e30e6999c25 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Mon, 6 May 2024 19:00:16 +0300
Subject: [PATCH 2/2] Update my name to the preferred format in authors
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Timo Teräs <timo.teras@iki.fi>
---
 AUTHORS | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/AUTHORS b/AUTHORS
index 34a85e2a1..e4dc9fdc8 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -35,7 +35,7 @@ Phil Vandry <vandry@tzone.org>
 pnd23@telenet.be <pnd23@telenet.be>
 P. Pronk <xfce@pronk.nl>
 Thomas Leonard <tal@ecs.soton.ac.uk>
-Timo <timo.teras@iki.fi>
+Timo Teräs <timo.teras@iki.fi>
 Viktor Odintsev <ninetls@xfce.org>
 
 Credits :
-- 
GitLab

